### PR TITLE
Updating code_utils.py to solve issue #1747

### LINKED ISSUE #1747: [Bug]: "Neither powershell nor pwsh is installed." always printed when code_utils imported
This print is noisy. It should be sent to the logger instead of just printed.

https://github.com/microsoft/autogen/blob/a34e4cc515ae85c877a5e1ea59cc0bacf1320cd9/autogen/code_utils.py#L232

For example
```python
logger.info(...)
```

Introduced in #1710

### PR DESCRIPTION
**Pull request by abhaymathur21**
## Why are these changes needed?

These changes are needed since the print statement  print("Neither powershell nor pwsh is installed.") was being noisy and unnecessarily repetitive. It was giving this warning for linux systems as well thinking that there is something wrong with the system because neither powershell nor pwsh is installed but linux systems dont have either anyway so this warning is unnecessary for non-windows operating systems.

I added a condition so that the warning only gets issued if the OS is detected as windows since if it is any other OS, powershell or pwsh need not be there, some other terminal may be used like bash for linux for example.

image

I also replace print with logging.warning so that the warning isnt given repetitively 

## Related issue number

Closes #1747 

## Checks

- [x] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [x] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [x] I've made sure all auto checks have passed.

### CODE REVIEW

**Review by jackgerrits — APPROVED** [review]
Thanks!

File: autogen/code_utils.py [hunk 1]
```diff
@@ -229,8 +229,11 @@ def get_powershell_command():
                 return "pwsh"
 
         except FileNotFoundError:
-            print("Neither powershell nor pwsh is installed.")
-            return None
+            if WIN32:
+                logging.warning("Neither powershell nor pwsh is installed but it is a Windows OS")
+                return None
```
Comment (BeibinLi) [inline autogen/code_utils.py:234]: The code can be simplified. We only need one "return" here.

```
if WIN32:
                logging.warning("Neither powershell nor pwsh is installed but it is a Windows OS")
return None
```

File: test/test_code_utils.py [hunk 2]
```diff
@@ -574,21 +574,29 @@ def test_get_powershell_command_pwsh(self, mock_subprocess_run):
 
         self.assertEqual(get_powershell_command(), "pwsh")
 
+        # mock_subprocess_run.side_effect = [FileNotFoundError, MagicMock(return_value=MagicMock(returncode=0, stdout=StringIO("7")))]
```
Comment (BeibinLi) [inline test/test_code_utils.py:577]: Let's remove the outdated code (comments) here.

**Review by BeibinLi — COMMENTED** [review]
Thanks so much for the PR! I made two lint comments regarding the changes.

Happy to approve this PR anytime.

File: test/test_code_utils.py (hunk 2 above)
Comment (abhaymathur21) [reply test/test_code_utils.py]: Done!

**Review by BeibinLi — APPROVED** [review]
LGTM~

Thanks for resolving the issue!

**Comment by abhaymathur21** [comment]
> LGTM~
> 
> Thanks for resolving the issue!

No problem! How do I merge this PR now?

**Pull request merged**